genai-perf/genai_perf/inputs/retrievers/synthetic_prompt_generator.py

```diff
@@ -175,17 +174,25 @@ def _generate_prompt_with_token_reuse(
         final_prompt: List[int] = []
         size_to_use = block_size
         for index, hash_index in enumerate(prompt_hash_list):
-            if index == len(prompt_hash_list) - 1:
-                size_to_use = num_tokens - (index * block_size)
             if hash_index not in cls._cache:
-                # To ensure that the prompt doesn't merge chunks, we pop the last token
-                # and insert the bos token at the beginning. Length is maintained and
-                # the prompt generates the expected number of tokens.
+                # To ensure that the prompt doesn't merge chunks, we pop the
+                # last token and insert the bos token at the beginning. Length
+                # is maintained and the prompt generates the expected number
+                # of tokens.
                 prompt_tokens = cls._generate_prompt_tokens(size_to_use)
                 prompt_tokens.pop(-1)
                 prompt_tokens.insert(0, tokenizer.bos_token_id())
                 cls._cache[hash_index] = prompt_tokens
-            final_prompt.extend(cls._cache[hash_index])
+            if num_tokens <= 0:
+                raise ValueError(
+                    "Internal error: number of tokens should never be negative."
+                )
```

Comment:
nit: error message says negative, but if statement includes zero case

Reply:
Thanks for catching this. I was updating the error message in response to feedback above and accidentally added this.